sensor/common/connection/connection.go

```diff
@@ -0,0 +1,137 @@
+package connection
+
+import (
+	"context"
+	"crypto/x509"
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/pkg/errors"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/grpc/util"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/sensor/common/centralclient"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// This package is responsible for establishing a gRPC connection between sensor
+// and Central. The code here was previously part of sensor. Sensor used to
+// receive an HTTP client as a parameter and create the gRPC connection internally.
+// The idea here is to extract this behavior to outside of sensor so mocking becomes
+// easier.
+
+type ConnectionFactory interface {
+	SetCentralConnectionWithRetries(ptr *util.LazyClientConn)
+	StopSignal() concurrency.ErrorSignal
+	OkSignal() concurrency.Signal
+}
+
+
+type connectionFactoryImpl struct {
+	endpoint   string
+	httpClient *centralclient.Client
+
+	stopSignal concurrency.ErrorSignal
+	okSignal   concurrency.Signal
+}
+
+func NewConnectionFactor(endpoint string) (*connectionFactoryImpl, error) {
```

Comment:
```suggestion
func NewConnectionFactory(endpoint string) (*connectionFactoryImpl, error) {
```

Reply:
Thanks! Refactored in a separate commit.